client/src/main/java/org/apache/uniffle/client/impl/ShuffleWriteClientImpl.java

```diff
@@ -1057,6 +1076,21 @@ private void throwExceptionIfNecessary(ClientResponse response, String errorMsg)
     }
   }
 
+  private void throwExceptionIfNecessary(
+      ClientResponse response, String errorMsg, int stageAttemptNumber) {
+    if (response != null && response.getStatusCode() != StatusCode.SUCCESS) {
+      if (stageAttemptNumber > 0) {
```

Comment:
This method throws a NotRetryException on Stage retry. @zuston

Reply:
Oh no, my thought is to fast fail when encountering the outdate send request.

Reply:
> Oh no, my thought is to fast fail when encountering the outdate send request.

OK,Have been changed.